scripts/makeheader

```diff
@@ -3,9 +3,41 @@ from __future__ import print_function
 import re
 import os
 
-# expected format:
-# #define XF86XK_FooBar _EVDEVK(0x123) /* some optional comment */
-evdev_pattern = re.compile(r'^#define\s+XF86XK_(?P<name>\w+)\s+_EVDEVK\((?P<value>0x[0-9A-Fa-f]+)\)')
+# Expected format:
+#     #define XF86XK_FooBar 0x1234         /* some optional comment */
+# or:
+#     #define XF86XK_FooBar _EVDEVK(0x123) /* some optional comment */
+# We also need to match commented evdev entries:
+#     /* Use: XF86XK_FooBar _EVDEVK(0x123)    some optional comment */
+keysym_entry_pattern = re.compile(
+    r'''^
+    (?P<define>\#define|/\*\s+Use:)\s+
+    (?P<prefix>\w*)XK_(?P<name>\w+)(?P<spacing>\s+)
+    (?P<evdev>_EVDEVK\()?(?P<value>0x[0-9A-Fa-f]+)(?(evdev)\))
+    ''',
+    re.VERBOSE,
+)
+# Match remaining XK_ references
+xorgproto_keysym_prefix_pattern = re.compile(r'\b(?P<prefix>\w*)XK_(?!KOREAN\b)')
+
+
+def make_keysym_entry(m: re.Match[str]) -> str:
+    '''
+    Perform the substitutions
+    '''
+    if m.group('evdev'):
+        if m.group('define').startswith('#'):
+            # Replace the xorgproto _EVDEVK macro with the actual value
+            # 0x10081000 is the base, the evdev hex code is added to that.
+            # We replace to make parsing of the keys later easier.
```

Comment:
fwiw, this comment is now confusing since we're no longer replacing the line

Reply:
Sorry, I do not understand:  we _do_ replace the macro when it is not commented.